### PR TITLE
chore: add IRSA usage doc

### PR DESCRIPTION
**Pull request by JBOClara**
**What this PR does**:

Adding doc about IAM role usage.

**Which issue(s) this PR fixes**:
Complete #1152

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [X] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [X] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by adejanovski — APPROVED** [review]
Thanks a lot @JBOClara , much appreciated!

File: docs/content/en/tasks/backup-restore/_index.md [hunk 1]
```diff
@@ -155,6 +155,42 @@ spec:
...
+      prefix: demo
+      credentialsType: role-based
+      storageSecretRef:
+        name: ""
+    ...
+```
+
+To make this work, you must ensure the following steps are completed:
+
+> While Medusa is running in standalone mode, it uses the default service account from the namespace. Make sure this service account has the necessary role annotation.
```
Comment (c3-clement) [inline docs/content/en/tasks/backup-restore/_index.md:184]: Is this behavior acceptable ?
That means that the `default` service account needs to be properly annotated and needs to be properly bound with the AWS IAM role. 

Therefore, any pods created without a SA specified will be granted with AWS permissions.
This against the least privilege principle - many of our customers won't accept this.

Comment (c3-clement) [reply docs/content/en/tasks/backup-restore/_index.md:184]: Following this, just opened this issue https://github.com/k8ssandra/k8ssandra-operator/issues/1294

**Comment by adejanovski** [comment]
FYI folks, [this](https://github.com/k8ssandra/k8ssandra-operator/issues/1066) is in our current dev cycle. We'll probably end up removing this pod altogether since its original use case is no longer valid, and the one we had planned for (bootstrapping a new cluster using a backup) isn't implemented yet.

**Pull request merged**